react/features/base/participants/reducer.js

```diff
@@ -147,31 +142,67 @@ function _participant(state: Object = {}, action) {
 }
 
 /**
- * Listen for actions which add, remove, or update the set of participants in
- * the conference.
+ * Reduces a specific redux action of type {@link PARTICIPANT_JOINED} in the
+ * feature base/participants.
  *
- * @param {Participant[]} state - List of participants to be modified.
- * @param {Object} action - Action object.
- * @param {string} action.type - Type of action.
- * @param {Participant} action.participant - Information about participant to be
- * added/removed/modified.
- * @returns {Participant[]}
+ * @param {Object} state - The participant element of the redux state of the
+ * feature base/participants. Given that the redux action is of type
+ * {@code PARTICIPANT_JOINED}, it is expected that {@code state} is an empty
+ * object.
+ * @param {Action} action - The redux action of type {@code PARTICIPANT_JOINED}
+ * to reduce.
+ * @private
+ * @returns {Object} The next/new participant element of the redux state of the
+ * feature base/participants after the reduction of the specified
+ * {@code action}.
  */
-ReducerRegistry.register('features/base/participants', (state = [], action) => {
-    switch (action.type) {
-    case DOMINANT_SPEAKER_CHANGED:
-    case PARTICIPANT_ID_CHANGED:
-    case PARTICIPANT_UPDATED:
-    case PIN_PARTICIPANT:
-        return state.map(p => _participant(p, action));
-
-    case PARTICIPANT_JOINED:
-        return [ ...state, _participant(undefined, action) ];
+function _participantJoined(state, { participant }) {
+    const {
+        avatarURL,
+        connectionStatus,
+        dominantSpeaker,
+        email,
+        isBot,
+        local,
+        name,
+        pinned,
+        role
+    } = participant;
+    let { avatarID, conference, id } = participant;
+
+    if (local) {
+        // avatarID
+        //
+        // TODO Get the avatarID of the local participant from localStorage.
+        avatarID || (avatarID = randomHexString(32));
 
-    case PARTICIPANT_LEFT:
-        return state.filter(p => p.id !== action.participant.id);
+        // conference
+        //
+        // XXX The local participant is not identified in association with a
+        // JitsiConference because it is identified by the very fact that it is
+        // the local participant.
+        conference = undefined;
 
-    default:
-        return state;
+        // id
+        //
+        // XXX The situation of not having an ID for a remote participant should
```

Comment:
I know you're just moving code around, but since we are here, let's fix it: this is the local participant, so I think the comment can go away.

Reply:
OK. Will come in "Associate remote participant w/ JitsiConference (_JOINED)".